src/olympia/reviewers/tests/test_views.py

```diff
@@ -8095,17 +8109,39 @@ def setUp(self):
             needs_admin_code_review=True,
         )
 
+        # Mixed listed and unlisted versions. Should not show up in queue.
+        mixed_addon = addon_factory()
+        VersionReviewerFlags.objects.create(
+            version=version_factory(addon=mixed_addon,
+                                    channel=amo.RELEASE_CHANNEL_UNLISTED,
+                                    created=self.days_ago(5)),
+            needs_human_review_by_mad=False
+        )
+        VersionReviewerFlags.objects.create(
+            version=version_factory(addon=mixed_addon,
+                                    channel=amo.RELEASE_CHANNEL_LISTED,
+                                    created=self.days_ago(3)),
+            needs_human_review_by_mad=True
+        )
+        VersionReviewerFlags.objects.create(
+            version=version_factory(addon=mixed_addon,
+                                    channel=amo.RELEASE_CHANNEL_LISTED,
+                                    created=self.days_ago(1)),
+            needs_human_review_by_mad=False
+        )
+
         self.expected_addons = [listed_addon, unlisted_addon]
 
     def test_results(self):
-        response = self.client.get(self.url)
+        with self.assertNumQueries(30):
```

Comment:
```suggestion
        with self.assertNumQueries(30):
            # 30 queries is a lot. Some of them are unfortunately scaling with
            # the number of add-ons in the queue.
            # - 2 for savepoints because we're in tests
            # - 2 for user/groups
            # - 11 for various queue counts, including current one
            #      (unfortunately duplicated because it appears in two
            #       completely different places)
            # - 3 for the addons in the queue and their files (regardless of
            #     how many are in the queue - that's the important bit)
            # - 2 for config items (motd / site notice)
            # - 2 for my add-ons / my collection in user menu
            # - 4 for reviewer scores and user stuff displayed above the queue
            # - 2 queries for first add-on to get listed/unlisted count of
            #     versions with needs human review flag
            # - 2 queries for second add-on to get listed/unlisted count of
            #     versions with needs human review flag
```